velox/exec/MergeJoin.cpp

```diff
@@ -1141,7 +1178,7 @@ RowVectorPtr MergeJoin::doGetOutput() {
           isFullJoin(joinType_)) {
         // If output_ is currently wrapping a different buffer, return it
         // first.
-        if (prepareOutput(input_, nullptr)) {
+        if (prepareOutput(input_, rightInput_)) {
```

Comment:
Passing a `nullptr `to `prepareOutput `results in generating more small batches, as the `nullptr `does not match `currentRight_`. This leads to additional overhead due to extra copying in the `mergeOutputBatches `method.